azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageTypeServlet.java

```diff
@@ -74,10 +82,36 @@ protected void handleGet(
       throws ServletException, IOException {
     try {
       final Map<String, String> templateVariableToValue = new HashMap<>();
-      if (IMAGE_TYPE_WITH_ID_URI_TEMPLATE.match(req.getRequestURI(), templateVariableToValue)) {
+      ImageTypeDTO imageTypeDTO = null;
+      if (StringUtils.isNotBlank(req.getPathInfo())) {
+        String imageTypeName = req.getPathInfo().substring(INDEX_AFTER_LEADING_SLASH);
+        if (imageTypeName.contains(PATH_SEPARATOR)) {
```

Comment:
Why is an explicit check needed for the path separator in the imageTypeName and are imageTypeNames even allowed to have a separator ?

Reply:
req.getPathInfo() will return whatever is after the imageTypes/ and before the first parameter, IE the "?". So if someone calls /imageType/a/imageType?proxyUser=proxyUser, we will get /a/imageType for req.getPathInfo(). This is clearly not a valid imageType. So we can either have execution continue, and have it fail gracefully do to that image type not being in the database, or we can check directly. Do you have a preference?

Reply:
Got it, thanks for the explanation. For this case I would recommend having the check refactored into a utility method, something like `validatedImageTypeName(String name)` 
That will allow (1) easy updates/additions to the checks (2) reusability of the same from other parts of the code.